store/auth.go

```diff
@@ -198,3 +199,67 @@ func RequestPackageAccessMacaroon() (string, error) {
 	}
 	return responseData.Macaroon, nil
 }
+
+// RequestDischargeMacaroon requests a discharge macaroon for an existing macaroon
+func RequestDischargeMacaroon(username, password, macaroon, otp string) (string, error) {
+	const errorPrefix = "cannot get discharge macaroon: "
+
+	data := map[string]string{
+		"email":    username,
+		"password": password,
+		"macaroon": macaroon,
+	}
+	if otp != "" {
+		data["otp"] = otp
+	}
+	dischargeJSONData, err := json.Marshal(data)
+	if err != nil {
+		return "", fmt.Errorf(errorPrefix+"%v", err)
+	}
+
+	req, err := http.NewRequest("POST", ubuntuoneDischargeAPI, strings.NewReader(string(dischargeJSONData)))
+	if err != nil {
+		return "", fmt.Errorf(errorPrefix+"%v", err)
+	}
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("content-type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf(errorPrefix+"%v", err)
+	}
+	defer resp.Body.Close()
+
+	// check return code, error on 4xx and anything !200
+	switch {
+	case httpStatusCodeClientError(resp.StatusCode):
+		// get error details
+		var msg ssoMsg
+		dec := json.NewDecoder(resp.Body)
+		if err := dec.Decode(&msg); err != nil {
+			return "", fmt.Errorf(errorPrefix+"%v", err)
+		}
+		if msg.Code == "TWOFACTOR_REQUIRED" {
+			return "", ErrAuthenticationNeeds2fa
+		}
+
+		return "", fmt.Errorf(errorPrefix+"%v", msg.Message)
```

Comment:
```
if msg.Message != "" {
        return "", fmt.Errorf(errorPrefix+"%v", msg.Message)
}
fallthrough
```

This will make it go into the generic error message if it cannot find a proper message to display.